fix: throw a useful error if rendering undefined entry (#12113)
withastro/astro: .changeset/wise-pumas-fry.md

---
'astro': patch
---

> Throw a helpful error if attempting to render undefined entry

Reviewer: ```suggestion
Adds a helpful error when attempting to render an undefined collection entry
```
